### ISSUE TITLE
Frontend Usage Overloads the Server: Wordpress Connection Lost Error

### ISSUE THREAD

**Issue by esilverman682** [wpengine/faustjs#827]
Clicking links such as the header navigation causes an overload on the server which then in turn slows down the headless Wordpress  dashboard with a "Connection Lost" message. 

Currently using:
- Nextjs
- Headless Wordpress 
- Faustjs plugin
- Using Static Props and revalidate is at 60.
- CPU & Mem from host is more than enough to handle the WP Instance

It seems like calls are being made to the backend even though no Server Side Props are performed in the frontend.

**Comment by theodesp** [comment]
@esilverman682 are you using the starter project because it tries to connect to the Wordpress WPGraphQL API to fetch data during build time?

**Comment by esilverman682** [comment]
Yes. The only server side calls would be for forms for gets and posts.

**Comment by blakewilson** [comment]
@esilverman682 👋 

Could you examine the Network tab in the Developer Tools in your browser and see if there are any requests to the `/graphql` endpoint in WordPress? It's possible there is some `useEffect` or an infinite loop that is making requests to WordPress, and crashing the server.

**Comment by esilverman682** [comment]
Great point. It looks like it is calling any image on the frontend page from the Wordpress backend. Is there any way for the images to just be called at build time instead of this way. ( I set up the blog with faustjs out the box  and it seems to be calling all images as well). Any suggestions are greatly appreciated.

**Comment by theodesp** [comment]
@esilverman682 calling images from the client means that it does a lot of Client Side Requests. If you are serving images, it's better if you use the  [next image](https://nextjs.org/docs/api-reference/next/image) component. For example take a look a this blueprint portfolio page which is build using Faust.js:

[Portfolio page](https://ho9og0bwp1wgbx352zgxg3euj.js.wpenginepowered.com/project)

Try to scroll down while inpecting the dev tab, You will see that each image loads on demand as it comes into view:
<img width="1786" alt="Screenshot 2022-04-06 at 08 36 58" src="https://user-images.githubusercontent.com/328805/161921067-e11677cc-1b43-40a1-853d-19795fb2e618.png">

The code for this component is here if you want to take a look:

https://github.com/wpengine/atlas-blueprint-portfolio/blob/main/src/components/FeaturedImage/FeaturedImage.js

**Comment by esilverman682** [comment]
I have all images working with the next "Image" tag. Do you mind taking a quick look to see if you can spot anything. It seems to make one initial graphql call and calls a cluster of images once for the blog and once for the other pages. I also implemented dynamic imports after the banner on each page.  Im wondering if the backend ACFs and the frontend queries have something to do with it or perhaps the preview feature or revalidate..not quite sure. I would appreciate your expert opinion. url --> kap2-revised.vercel.app

**Comment by theodesp** [comment]
@esilverman682 That is really strange indeed, because the `next/image` should lazy load. For example I modified the example project we have for posts and added the `next/image` for the featured images and it lazy loads them:

```tsx
import React from 'react';
import Link from 'next/link';
import Image from 'next/image';
import type { Post } from 'client';
import styles from 'scss/components/Posts.module.scss';
import Heading, { HeadingProps } from './Heading';

interface Props {
  posts: Post[] | undefined;
  intro?: string;
  id?: string;
  heading?: string;
  headingLevel?: HeadingProps['level'];
  postTitleLevel?: HeadingProps['level'];
  readMoreText?: string;
}

function Posts({
  posts,
  intro,
  heading,
  id,
  headingLevel = 'h1',
  postTitleLevel = 'h2',
  readMoreText = 'Read more',
}: Props): JSX.Element {
  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <section className={styles['posts-block']} {...(id && { id })}>
      <div className="wrap">
        {heading && (
          <Heading level={headingLevel} className={styles.heading}>
            {heading}
          </Heading>
        )}
        {intro && <p className={styles.intro}>{intro}</p>}
        <div className="posts">
          {posts.map((post) => {
            const image = post?.featuredImage?.node?.sourceUrl();
            return (
            <div
              className={styles.single}
              key={post.id ?? ''}
              id={`post-${post.id}`}>
              <div>
                <Heading level={postTitleLevel} className={styles.title}>
                  <Link href={`/posts/${post.slug}`}>
                    <a>{post.title()}</a>
                  </Link>
                </Heading>
                <div>
                  {image && <Image src={image} width={340} height={340}/>}
                </div>
                <div
                  className={styles.excerpt}
                  // eslint-disable-next-line react/no-danger
                  dangerouslySetInnerHTML={{ __html: post.excerpt() ?? '' }}
                />
                <Link href={`/posts/${post.slug}`}>
                  <a aria-label={`Read more about ${post.title || 'the post'}`}>
                    {readMoreText}
                  </a>
                </Link>
              </div>
            </div>
            )})}
          {posts && posts?.length < 1 && <p>No posts found.</p>}
        </div>
      </div>
    </section>
  );
}

export default Posts;
```
Could you provide a sample code for the Blog component so we can reproduce from our end?

**Comment by wrivkap** [comment]
Your suggestion for Image tags has worked. No more images called from the backend. There is a one-time initial graphql call from the faustjs frontend to the backend. Is this supposed to happen?

**Comment by theodesp** [comment]
Yes. Ideally GQty will aggregate all the results one query. The results for the featured images would be image urls for example:

`http://example.com/wpcontent/2020/01/01/image.png`

Then you feed it into `next/image` component:
`<Image src={image} width={340} height={340}/>`

which will perform the actual lazy load mechanism when the user visits the page it will try to request the image on demand.

**Comment by wrivkap** [comment]
Wordpress Faustjs Server Slowdown 
1. I have an average of 10 ACFs fields per page. Could this also create a slowdown on the Wordpress end?
<img width="1285" alt="Screen Shot 2022-04-13 at 6 42 15 PM" src="https://user-images.githubusercontent.com/101981793/163282555-90313713-1b78-4a01-a052-1f61e5e3fff2.png">

The permalink and preview button on Wordpress work for the Faust frontend.
1. When I set to Private Status. It hides the content on the frontend as it should
2. When I set it back to Publish it doesnt change the status on the frontend unless I rebuild the whole application again.

is there a way for the Nextjs (faust) frontend to pickup on the updated Private to Public status without rebuilding?
<img width="321" alt="Screen Shot 2022-04-11 at 11 39 37 PM" src="https://user-images.githubusercontent.com/101981793/163282588-2f5126aa-5b3f-4f3a-9c60-33b100e90a71.png">
